src/fund/utils/subscribeToWindowMessage.ts

```diff
@@ -0,0 +1,72 @@
+import { DEFAULT_ONRAMP_URL } from '../constants';
+import type { JsonObject } from '../types';
+
+export enum MessageCodes {
+  AppParams = 'app_params',
+  PaymentLinkSuccess = 'payment_link_success',
+  PaymentLinkClosed = 'payment_link_closed',
+  GuestCheckoutRedirectSuccess = 'guest_checkout_redirect_success',
+  Success = 'success',
+  Event = 'event',
+}
+
+export type MessageCode = `${MessageCodes}`;
+
+export type MessageData = JsonObject;
+
+export type PostMessageData = {
+  eventName: MessageCode;
+  data?: MessageData;
+};
```

Comment:
do these types need to be exported? didn't see them used anywhere else

Reply:
Not needed to be exported. Removed `export`